In reference to issue #126 search option enhancement

nikhilrayaprolu opened the pull request:
In reference to issue #126,
started working on advanced component, since it is a large issue with multiple small tasks that need to be done, different implementations are required for different inputs, solving this issue iteratively.
this is the pull with iteration 1:-
![image](https://cloud.githubusercontent.com/assets/15216503/25903238/76e4d308-35b9-11e7-802e-fd4f343089c7.png)

google's advanced search:-

![image](https://cloud.githubusercontent.com/assets/15216503/25904367/adf97b7a-35bc-11e7-82a7-4dab95441cd2.png)
demo link available at https://susper-pr-255.herokuapp.com/

nikhilrayaprolu commented:
implemented the first part of #126 will continue do the 2nd and 3rd on next pulls.

mariobehling commented:
Please avoid new colors and keep it simple. Simply follow Google in design for now, please.

nikhilrayaprolu commented:
I have updated the picture to show google's advanced search
sure I will check with the alignments.
secondly, @harshit98 you were complaining about writing the component in each file. I didn't face that problem you could check the test files I updated.

harshit98 commented:
@nikhilrayaprolu Yes I saw that. I'll follow the same next time. Good work!

harshit98 commented:
@nikhilrayaprolu Please fix the color of `Advanced search` as well. It is not matching Google design.

nikhilrayaprolu commented:
@harshit98 squashed :smile:

harshit98 commented:
@nikhilrayaprolu Please go through, other suggestions as well.

nikhilrayaprolu commented:
@harshit98  done with the changes

nikhilrayaprolu commented:
@harshit98  done with the changes.

harshit98 commented on src/app/footer-navbar/footer-navbar.component.css:
```diff
@@ -8,8 +8,6 @@
   align-items: center;
   margin-bottom: 0px;
-  position: fixed;
-  bottom: 0px;
```
@nikhilrayaprolu why you removed those lines? They fixed the issue #244

nikhilrayaprolu replied:
The position attributes should be independent of the child component I inherit, they should depend only on parent component. when I use the same component in other parent components, I should not have any internal css factors of the child component affecting the position in the parent component, there should be independency to apply my own css  in each parent component I inherit this. so where ever you want to fix the component as fixed , write this css only in the parent component but not in the child component, such that we could have other positions defined in other parent components.
